### ISSUE TITLE
Could you please add relation - one file to many albums

### ISSUE THREAD

**Issue by ghost** [LycheeOrg/Lychee#1095]
Hi, I would like to have ability to assign one file to many albums / folders and I'm not thinking about smart folders. Let's say, that I have photo called "photo1" and I have "folder1" and "folder2". I would like to be able to assign that photo to both of the folders.

**Comment by kamil4** [comment]
While you can't currently have the same `Photo` object belong to multiple albums, you can duplicate an existing photo to a different album (using `Copy to...` from the menu that pops up when you right-click while hovering over a photo thumbnail). Only the database row will get duplicated; the image files on disk will be shared between the two photos. Is that good enough for you?

**Comment by nagmat84** [comment]
As one knows, I am a strong advocate of this enhancement. I have already proposed that enhancement myself in the past. My reasons are not because it would save storage (it won't as @kamil4 has already pointed out), but because it would save us some complexity in the code and because it would be the theoretically clean approach. At the moment the code which decides if a file may be deleted or not is more complex than necessary, because we have to manually determine first whether the photo is used by more than one album. Also, some attributes of "copies" of the identical photo may differ depending on the album the copy of the photo belongs to which leads to inconsistencies and hence should be fixed.

As you also know, @ildyria does not like the idea of a (m:n)-relation between photos and albums, although I have never fully understood his reasons.

However, although I am a strong fan of this enhancement and although I am also willing to implement it, I discourage to do it **now**. First, the fundamental codebase needs a proper cleanup and the things which are already there should be re-factored, before we change this aspect. But then I would really consider doing it. IMHO, I might be ready for it in half a year.

In summary, I would leave this request for enhancement open until then.

**Comment by ildyria** [comment]
Work in progress.
